base/mysqlxx/Pool.h

```diff
@@ -189,7 +189,7 @@ class Pool final
     ~Pool();
 
     /// Allocates connection.
-    Entry get();
+    Entry get(uint64_t wait_timeout);
```

Comment:
Why we can't reuse `connect_timeout_` that is passed into Pool constructor ?
Also why this `wait_timeout` is not passed into constructor like other options ?

Reply:
>Why we can't reuse connect_timeout_ that is passed into Pool constructor ?

I want to distinct `connect_timeout` from `wait_timeout`, since you may want to set it to 0 to avoid waiting for free connections at all, while you can't set connect timeout to 0 (also with `connect_timeout` only `connection_pool_size` can wait, while with `connection_wait_timeout` you can have tons of waiters, by number of active queries)

>Also why this wait_timeout is not passed into constructor like other options ?

I wasn't though about this at first, but It looks more natural to see the timeout for get instead of where the pool was created. Consider `MaterializedMySQLSyncThread` it uses `pool->get()` but the pool had been created in `DatabaseFactory`

Reply:
@azat thanks for answer. Make sense.